spaceship/lib/spaceship/connect_api/api_client.rb

```diff
@@ -159,15 +165,15 @@ def with_asc_retry(tries = 5, &_block)
 
         if [500, 504].include?(status)
           msg = "Timeout received! Retrying after 3 seconds (remaining: #{tries})..."
-          raise msg
+          raise TimeoutError, msg
```

Comment:
Created new error class so this can be specifically caught and handled in the retry logic

Reply:
Although it's deprecated there is exactly the same error class in the standard library in Ruby. So it would be great to prefix here with the namespace to not confuse Ruby or rename the own error class🙂 

https://rubyapi.org/3.0/o/object/timeouterror

Reply:
Oh yikes! I’ll just rename it lol

Reply:
@ainame Updated it! Renamed it to `TimeoutRetryError` since... that’s kind of what it is 🤷‍♂️